scripts/pre_commit_linter.py

```diff
@@ -468,6 +468,59 @@ def _pre_commit_linter(all_files):
     return summary_messages
 
 
+def _check_newline_character(all_files):
+    """This function is used to check that each file
+    ends with a single newline character.
+    """
+    print 'Starting newline-at-EOF checks'
+    print '----------------------------------------'
+    total_files_checked = 0
+    total_error_count = 0
+    summary_messages = []
+    all_files = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    failed = False
+    for filename in all_files:
+        with open(filename, 'rb') as f:
+            total_files_checked += 1
+            total_num_chars = 0
+            for line in f:
+                total_num_chars += len(line)
+            if total_num_chars == 1:
+                failed = True
+                print '%s --> Error: Only one character in file' % filename
+                total_error_count += 1
+            elif total_num_chars > 1:
+                f.seek(-2, 2)
+                if not (f.read(1) != '\n' and f.read(1) == '\n'):
+                    failed = True
+                    print '%s --> Please ensure that this file ends \
```

Comment:
OK -- please let me know when you've pushed the code; I can't review until you've done so.